R/utils-testopt.R

```diff
@@ -210,7 +210,10 @@ test_optim <- function(optim, ...,
     y <- torch::torch_tensor(y0, requires_grad = TRUE)
 
     # instantiate optimizer
-    optim <- do.call(optim, c(list(params = list(x, y)), opt_hparams))
+    if(!missing(opt_hparams))
```

Comment:
```suggestion
    if(!is.null(opt_hparams))
```

`missing` checks if the argument has been specified or not. Since there's a default of `NULL`, I think it will always say `FALSE`.

Reply:
Well, i tested for that:

```
f <- function(x = NULL) {      
  return(missing(x))
}

f()

1] TRUE

```